lib/iex/lib/iex/introspection.ex

```diff
@@ -637,8 +637,9 @@ defmodule IEx.Introspection do
       {:ok, types} ->
         printed =
           for {_, {^type, _, args}} = typespec <- types do
-            doc = {format_type(typespec), type_doc(module, type, length(args))}
-            print_typespec(doc)
+            type_doc(module, type, length(args))
+            |> Tuple.insert_at(0, format_type(typespec))
+            |> print_typespec()
```

Comment:
It seems we end-up prepending `format_type/1` both times we call `type_doc/3` (here and below)... so what if we call `format_type/1` inside `type_doc/3` and skip this dance?